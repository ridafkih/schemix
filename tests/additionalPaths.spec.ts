import { createSchema } from "schemix";
import { describe, expect, it } from "vitest";

describe("index", () => {
  it("should allow compiling schemas from multiple folders with additionalPaths", async () => {
    const schema = createSchema({
      basePath: `${__dirname}/__schema__`,
      additionalPaths: ["../__additional_schema__/"],
      datasource: {
        provider: "cockroachdb",
        url: { env: "DATABASE_URL" },
      },
      generator: {
        provider: "prisma-client-js",
      },
    });

    const asString = await schema.toString();
    expect(asString).toMatchSnapshot();
  });
});
