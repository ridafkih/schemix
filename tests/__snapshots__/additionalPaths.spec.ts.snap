// Vitest Snapshot v1

exports[`index > should allow compiling schemas from multiple folders with additionalPaths 1`] = `
"datasource database {
  provider = \\"cockroachdb\\"
  url      = env(\\"DATABASE_URL\\")
}

generator client {
  provider = \\"prisma-client-js\\"
}

enum UserStatus {
  ACTIVE
  INACTIVE
  BANNED
}

enum Status {
  PENDING
  LIVE
  DELETED
  REMOVED
}

model Post {
  id        String   @default(uuid()) @id
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  content   String
  deleted   Boolean
  imageUrl  String
}

model ImagePost {
  id        String   @default(uuid()) @id
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  content   String
  deleted   Boolean
  imageUrl  String
}

model User {
  id        String     @default(uuid()) @id
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
  status    UserStatus
  posts     Post[]
}

model Author {
  id        Int    @id @default(autoincrement())
  email     String
  firstName String
  LastName  String
}

model Book {
  id            Int      @id @default(autoincrement())
  initializedAt DateTime @default(now())
  editedAt      DateTime @updatedAt
  status        Status
  text          String
}
"
`;
